flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -414,6 +422,78 @@ protected static String getManuallyRefreshStatement(
         return insertStatement.toString();
     }
 
+    private static ResultFetcher callDropMaterializedTableOperation(
+            OperationExecutor operationExecutor,
+            OperationHandle handle,
+            DropMaterializedTableOperation dropMaterializedTableOperation) {
+        ObjectIdentifier tableIdentifier = dropMaterializedTableOperation.getTableIdentifier();
+        CatalogMaterializedTable materializedTable =
+                getCatalogMaterializedTable(operationExecutor, tableIdentifier);
+
+        if (CatalogMaterializedTable.RefreshStatus.ACTIVATED
+                == materializedTable.getRefreshStatus()) {
+            ContinuousRefreshHandler refreshHandler =
+                    deserializeContinuousHandler(
+                            materializedTable.getSerializedRefreshHandler(),
+                            operationExecutor.getSessionContext().getUserClassloader());
+            // get job running status
+            JobStatus jobStatus = getJobStatus(operationExecutor, handle, refreshHandler);
+            if (!jobStatus.isTerminalState()) {
+                try {
+                    cancelJob(operationExecutor, handle, refreshHandler);
+                } catch (Exception e) {
+                    jobStatus = getJobStatus(operationExecutor, handle, refreshHandler);
+                    // throw exception if job is not terminal state
+                    if (!jobStatus.isTerminalState()) {
+                        throw new SqlExecutionException(
+                                String.format(
+                                        "Drop materialized table failed, current job status is %s.",
+                                        jobStatus),
+                                e);
+                    } else {
+                        LOG.warn(
+                                "Cancel continuous refresh job {} of materialized table {} occur exception, but job is terminal state, skip the cancel operation.",
```

Comment:
```
"An exception occurred while canceling the continuous refresh job {} for materialized table {},"
                                        + " but since the job is in a terminal state, skip the cancel operation."
```